pubsub.go

```diff
@@ -0,0 +1,33 @@
+package mainflux
+
+// MessagePublisher specifies a message publishing API.
+type MessagePublisher interface {
+	// Publishes message to the stream. A non-nil error is returned to indicate
+	// operation failure.
+	Publish(RawMessage) error
+}
+
+// MessageSubscriber specifies a message subscription API.
+type MessageSubscriber interface {
+	// Subscirbes to channel. A non-nil error is returned to indicate
+	// operation failure.
+	Subscribe(string, func(RawMessage)) (Subscription, error)
+}
+
+// MessagePubSub specifies a message publishing and subscription API.
+type MessagePubSub interface {
+	MessagePublisher
+	MessageSubscriber
+}
+
+// MessageBroker specifies a message broadcasting API.
+type MessageBroker interface {
+	MessagePubSub
+	// Broadcasts raw message to channel.
+	Broadcast(RawMessage, func(RawMessage) error) error
+}
+
+// Subscription specifies subscription interface.
+type Subscription interface {
+	Unsubscribe() error
```

Comment:
I suggested that you share a structure. Subscription interface in this form is not that useful.